pkg/kubelet/apis/deviceplugin/v1alpha/api.proto

```diff
@@ -32,6 +32,8 @@ message RegisterRequest {
 	string endpoint = 2;
 	// Schedulable resource name. As of now it's expected to be a DNS Label
 	string resource_name = 3;
+        // Indicates if device initialization is required before each container start
+	bool init_required = 4;
```

Comment:
Since device plugins are not mature enough yet, we might want to start with timeout control with Kubelet. We can increase the timeout in kubelet and since changes involved will not be too big to pass the timeout control to plugins, in future we may enable that if need arises. This is my opinion.
@vishh @derekwaynecarr @dchen1107 @jiayingz what you guys think?

Reply:
+1 for starting with a const grpc timeout limit in kubelet and experiment with some light-weight health checking and recovery operations through this API. We understand on your side, you want to make sure gpu device is in good state for container to start, but remember applications in the container are also unavailable during this time window. If we have unbound container restart time, we may break some upper component assumptions.

Reply:
some more thoughts:

kubelet needs to have some sort of upper bound on grpc timeout. Just wanted to take that discussion a step further. I think we need both timeouts,
1. per device plugin timeout -- advertised by device plugins
2. Kubelet's upper bound default timeout 

- Device plugin registration will be failed if its timeout is more than Kubelet's upper bound.
- If plugin had advertised a within allowed range timeout, its own timeout will be used in the RPC
- If any plugin has not advertised timeout, Kubelet's default will be used in RPC

Why we want to have upper bound with kubelet is because there could be many device plugins running at the same time with random timeouts. It would be a nightmare for admin to manage delays without a upper bound at kubelet. With upper bound at kubelet, admin knows what max guaranteed delay could be there with RPC.

For now my PR has just kubelet upper bound timeout. I would like to do following things in a separate PR:
1. Make Kubelet device plugin upper bound timeout configurable using kubelet flag
2. Support device plugin advertised timeouts.